server/src-lib/Hasura/App.hs

```diff
@@ -156,6 +155,23 @@ mkPGLogger :: Logger Hasura -> Q.PGLogger
 mkPGLogger (Logger logger) (Q.PLERetryMsg msg) =
   logger $ PGLog LevelWarn msg
 
+-- | Context required for all graphql-engine commands
+data GlobalCtx
+  = GlobalCtx
+  { _gcHttpManager :: !HTTP.Manager
+  , _gcInstanceId  :: !InstanceId
```

Comment:
`_gcInstanceId` is only used by `serve`, right? And `InitCtx` is also only used by `serve`, right? If these two are right, then perhaps we can move `_gcInstanceId` to `InitCtx`, and rename `InitCtx` to something more fitting, such as `ServeCtx`.